pkg/ruler/scheduler.go

```diff
@@ -182,6 +184,18 @@ func (s *scheduler) poll() (map[string]configs.VersionedRulesConfig, error) {
 func (s *scheduler) addNewConfigs(now time.Time, cfgs map[string]configs.VersionedRulesConfig) {
 	// TODO: instrument how many configs we have, both valid & invalid.
 	level.Debug(util.Logger).Log("msg", "adding configurations", "num_configs", len(cfgs))
+	cycleNanos := float64(s.evaluationInterval.Nanoseconds())
+	nextEvalCycle := time.Unix(0, int64(math.Ceil(float64(now.UnixNano())/cycleNanos)*cycleNanos))
+	hasher := fnv.New64a()
+	evalOffset := func(userID string) time.Duration {
+		hasher.Reset()
+		hasher.Write([]byte(userID))
+		return time.Duration(int64(
+			math.Mod(
+				float64(hasher.Sum64()),
+				cycleNanos)))
+	}
```

Comment:
Why is this defined inline? Why not a method?

Reply:
Mostly due to where I put the hashing object, see the other comment.

Reply:
I would still lean to having a separate method that takes the hasher & `now`, or even a separate function that takes those two and `evaluationInterval`. Would make it much easier to test (hint hint).

Not going to insist on it though.